[Bug]: Reading default branch from wrong repo

### What happened?

When using the new token scheme (UoMResearchIT/RSE-Repository-Template#79) with a different repo, an error is generated. This is because the code to select the default branch of the repository is looking at the wrong repository (with a token that doesn't grant permission to look there). Workaround is an explicit origin branch name, but that shouldn't be necessary.

### What version of our actions are you running?

v1.2.3

### Which runner were you using?

ubuntu-latest

### Relevant log output

```shell
Run UoMResearchIT/actions/get-repo-ruleset@v1.2
  with:
    repository: UoMResearchIT/RSE-Repository-Template
    token: ***
Run $GITHUB_ACTION_PATH/fetch.bash
  $GITHUB_ACTION_PATH/fetch.bash
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    BRANCH: 
    GH_TOKEN: ***
    REPO_NAME: UoMResearchIT/RSE-Repository-Template
GraphQL: Could not resolve to a Repository with the name 'UoMResearchIT/Java-Web-Template'. (repository)
Error: Process completed with exit code 1.
```

--- a/get-repo-ruleset/fetch.bash
+++ b/get-repo-ruleset/fetch.bash
@@ -25,7 +25,7 @@ OUTPUT_FILE=$(realpath $OUTPUT_FILE.json)
 
 if [ -z "$BRANCH" ]; then
     # Fetch default branch dynamically
-    DEFAULT_BRANCH=$(gh repo view --json defaultBranchRef --jq .defaultBranchRef.name)
+    DEFAULT_BRANCH=$(gh repo view "$REPO_NAME" --json defaultBranchRef --jq .defaultBranchRef.name)
     echo "Default branch detected: $DEFAULT_BRANCH"
     BRANCH=$DEFAULT_BRANCH
 fi

